shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala

```diff
@@ -448,7 +454,7 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
   private def getColumnNames(
      selectedAttributes: Seq[Attribute],
      cacheAttributes: Seq[Attribute]): Seq[String] = {
-    selectedAttributes.map(a => "_col" + cacheAttributes.map(_.exprId).indexOf(a.exprId))
+    selectedAttributes.map(a => a.name)
```

Comment:
First off `cacheAttributes` is not used any more so it should be removed from the method signature. Second parquet has limits on the names it supports. It also does not want to have duplicate names, because reading it is ambiguous.  Spark does not have the same limitations. It is fine with duplicate names, because the expression id disambiguates them. Are we sure that this change will never run into limitations on parquet names?

Reply:
I didn't know about Spark not having a limitation on col names, thanks. The reason why I made the change was because now we are forcing people to set the column name on the column before writing parquet, but there's no guarantee there won't be duplicates. I will revert.